src/main/scala/ai/tripl/arc/transform/TypingTransform.scala

```diff
@@ -549,14 +551,22 @@ object Typing {
               val number = parseNumber(fmt, value)
               number.map( num => num.toString.toInt )
             }
-            case None => Option(value.toInt)
+            case None => {
+              if (value.toUpperCase.contains("E+")) {
```

Comment:
Other option is to just do the Big Decimal conversion then to an int, versus doing the conditional check.